### ISSUE TITLE
Latest release (1.0.0) breaks PyCharm run configuration

### ISSUE THREAD

**Issue by amureki** [Fapiko/intellij-better-direnv#10]
Greetings dear fellows,

thanks for the plugin. I am using latest PyCharm 2022.2.2 and latest plugin (1.0.0) breaks run configuration for me.
Instead of running tests or starting my dev server, it throws the next error:

```
class com.jetbrains.django.run.DjangoServerRunConfiguration cannot be cast to class com.jetbrains.python.run.PythonRunConfiguration (com.jetbrains.django.run.DjangoServerRunConfiguration and com.jetbrains.python.run.PythonRunConfiguration are in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @6e5fb29)
```

Small note - before that, version 0.3.0 was not throwing this error, however my direnv file was ignored (so the plugin was also not working).

Best,
Rust

**Comment by tejksat** [comment]
I have been investigating [the problem created in PyCharm issue tracker](https://youtrack.jetbrains.com/issue/PY-56364/IntelliJ-Python-Plugin-222416737-class-comjetbrainsdjangorunDjangoServerRunConfiguration-cannot-be-cast-to-class).

The problem seems to be in [this line](https://github.com/Fapiko/intellij-better-direnv/blob/47007205b2cc6b02d22965b7e0c1b2cd232e250a/modules/products/python/src/main/java/com/fapiko/jetbrains/plugins/better_direnv/runconfigs/PycharmEnvironmentProvider.java#L21). `PythonRunConfiguration` class is not the base class for all "pythonic" run configurations. Django, Flask, pytest and other run configurations do not extend to this class. As you need here `getCopyableUserData()` method, I would use `UserDataHolderBase` class, which defines this method. As an alternative, you may also use `AbstractPythonRunConfiguration` for this purpose. And I suggest to check if `pythonRunParams` can be cast to any of these classes with `instanceof` prior of the casting.

**Comment by jfreela** [comment]
This should be resolved by https://github.com/Fapiko/intellij-better-direnv/pull/12

The cast has been changed from PythonRunConfiguration to AbstractPythonRunConfiguration<?> which all py based run configs seem to use

**Comment by Fapiko** [comment]
Merged that PR and pushed a build today - should be published to the marketplace in a day or two.

**Comment by amureki** [comment]
> Merged that PR and pushed a build today - should be published to the marketplace in a day or two.

fantastic, thank you for the work! Will try it out later.

Cheers!

**Comment by tojoc** [comment]
I am getting the same issue in pycharm with the latest version (1.0.0) available for the plugin on PyCharm 2022.2.5 (Professional Edition)
> class com.jetbrains.python.testing.pytestconfiguration cannot be cast to class com.jetbrains.python.run.pythonrunconfiguration (com.jetbrains.python.testing.pytestconfiguration and com.jetbrains.python.run.pythonrunconfiguration are in unnamed module of loader com.intellij.ide.plugins.cl.pluginclassloader @50f0f863)
